### ISSUE TITLE
warning 204 only works on unused variables

### ISSUE THREAD

**Issue by Daniel-Cortez** [pawn-lang/compiler#479]
### Issue description:

Warning 204 (`symbol is assigned a value that is never used`) only works in simple cases when the variable is unused.
```Pawn
main()
{
	new n = 1; // warning 204: symbol is assigned a value that is never used: "n"
}
```
But if an unused value is assigned after the point where the variable is used, then the compiler won't warn about it:
```Pawn
main()
{
	new n = 1;
	printf("%d", n);
	n = 2; // no warning, even though this assignment is unused
}
```
Another problem is when the previously assigned value is unused, the compiler doesn't warn about this kind of errors at all:
```Pawn
new x = 1;
x = 2; // again, no warning, even though the previous value (1) is not used
```
Since the compiler already warns about unused assignments for unused variables, I think it would be beneficial to make it issue warnings in the aforementioned cases as well.

### Minimal complete verifiable example (MCVE):

See above.

### Workspace Information:

* Compiler version:
* Command line arguments provided (or sampctl version):
* Operating System:

**Comment by Daniel-Cortez** [comment]
Another special case:
```
new n = 0;
n = 1;
```
Here `new n = 0;` has the same effect as `new n;`, so it would probably make sense to inhibit the warning for such cases.

**Comment by Daniel-Cortez** [comment]
**To the maintainers: please do not close this issue yet.**
Even though #480 was merged, there's still some room for improvement. I have a couple of ideas on how to make the compiler detect more complicated cases of unused assignments, just need some time for implementation. If those ideas don't work out, I'll just close this issue myself.

**Comment by Daniel-Cortez** [comment]
All of the remaining ideas were implemented in #533, closing.
